File: 1D/data/TypesDeMateriauxAffichage.m

fichier=['TypesDeMateriaux.txt'];


fileID = fopen( fichier, 'r' );         % Ouverture du fichier
taille = str2num( fgetl( fileID ) );    % Dimensions de l'image
X=taille(1,1);
Y=taille(1,2);

t=0;
affichage=zeros(Y,X);
%prégénération de la matrice pour gangner du temps


for i=1:Y
    affichage(i,:)=str2num(fgetl( fileID ));
end

fclose(fileID);   

imagesc(affichage);title('Materiaux');

File: 1D/data/TypesDeMateriauxAffichage.py
import numpy as np
import matplotlib.pyplot as plt

fichier = "TypesDeMateriaux.txt"

with open(fichier, 'r') as fileID:
    taille = list(map(int, fileID.readline().split()))
    X, Y = taille[0], taille[1]

    t = 0
    affichage = np.zeros((Y, X))

    for i in range(Y):
        affichage[i, :] = list(map(float, fileID.readline().split()))

plt.figure()
plt.imshow(affichage, aspect='auto')
plt.title('Materiaux')
plt.savefig("Results/Materiaux.svg")
